src/lib/scheduler/worker.cpp

```diff
@@ -68,9 +69,13 @@ void Worker::_work() {
       }
     }
 
-    // Sleep if there is no ready task in our queue and work stealing was not successful.
+    // If there is no ready task neither in our queue nor in any other, worker waits for a new task to be pushed to the
+    // own queue or returns after timer exceeded (whatever occurs first).
     if (!work_stealing_successful) {
-      std::this_thread::sleep_for(std::chrono::milliseconds(10));
+      {
+        std::unique_lock<std::mutex> unique_lock(_queue->lock);
+        _queue->new_task.wait_for(unique_lock, std::chrono::microseconds(200));
```

Comment:
Could you move that into a `static constexpr` so that it stands out as a configuration parameter?

Example: `static constexpr auto DEFAULT_LIKE_SELECTIVITY = 0.1f;`